pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -718,6 +721,74 @@ const (
 	minSafeProxySize = 2
 )
 
+func (cr *PerconaXtraDBCluster) setProbesDefaults() {
+	if cr.Spec.PXC != nil {
+		if cr.Spec.PXC.LivenessInitialDelaySeconds != nil {
+			cr.Spec.PXC.LivenessProbes.InitialDelaySeconds = *cr.Spec.PXC.LivenessInitialDelaySeconds
+		} else if cr.Spec.PXC.LivenessProbes.InitialDelaySeconds == 0 {
+			cr.Spec.PXC.LivenessProbes.InitialDelaySeconds = 300
+		}
+
+		if cr.Spec.PXC.LivenessProbes.TimeoutSeconds == 0 {
+			cr.Spec.PXC.LivenessProbes.TimeoutSeconds = 5
+		}
+		if cr.Spec.PXC.LivenessProbes.SuccessThreshold > 1 || cr.Spec.PXC.LivenessProbes.SuccessThreshold == 0 {
+			cr.Spec.PXC.LivenessProbes.SuccessThreshold = 1
+		}
+
+		if cr.Spec.PXC.ReadinessInitialDelaySeconds != nil {
+			cr.Spec.PXC.ReadinessProbes.InitialDelaySeconds = *cr.Spec.PXC.ReadinessInitialDelaySeconds
+		} else if cr.Spec.PXC.ReadinessProbes.InitialDelaySeconds == 0 {
+			cr.Spec.PXC.ReadinessProbes.InitialDelaySeconds = 15
```

Comment:
I don't understand how code can reach this line
line 740 (not 742) will be executed `if cr.Spec.PXC.ReadinessProbes.InitialDelaySeconds == 0`

the same in HAProxy part

Reply:
We already have `readinessInitialDelaySeconds`  option in CR, and if this option is used, we set the old value  from it to new `InitialDelaySeconds` option.